RELEASE/scripts/autoscend/iotms/mr2017.ash

```diff
@@ -1,111 +1,119 @@
 	{
 		return false;
 	}
-	if(!canChangeToFamiliar(fam)) // Can Change to Familiar checks both whether we have it and if we are allowed to change to it
-	{
-		return false;
-	}
-	if(!auto_is_valid(fam))
-	{
-		return false;
-	}
-	familiar last = my_familiar();
 
+	bonus = to_lower_case(bonus);
+	// I don't want to alter CS behaviour so I'm leaving a couple things in that are otherwise irrelevant.
+	familiar last = my_familiar();
 	int goal = 0;
-	if((bonus == "meat") || (bonus == "1"))
-	{
-		goal = 1;
-	}
-	else if((bonus == "mp") || (bonus == "mp regen") || (bonus == "mpregen") || (bonus == "2"))
-	{
-		goal = 2;
-	}
-	else if((bonus == "muscle") || (bonus == "3"))
-	{
-		goal = 3;
-	}
-	else if((bonus == "item") || (bonus == "4"))
-	{
-		goal = 4;
-	}
-	else if((bonus == "mysticality") || (bonus == "myst") || (bonus == "5"))
+	
+	switch (bonus)
 	{
-		goal = 5;
+		case "1":
+		case "meat":
+			goal = 1;
+			break;
+		case "2":
+		case "mp":
+		case "mp regen":
+		case "mpregen":
+			goal = 2;
+			break;
+		case "3":
+		case "mus":
+		case "muscle":
+			goal = 3;
+			break;
+		case "4":
+		case "item":
+			goal = 4;
+			break;
+		case "5":
+		case "mysticality":
+		case "myst":
+			goal = 5;
+			break;
+		case "6":
+		case "hp":
+		case "hp regen":
+		case "hpregen":
+			goal = 6;
+			break;
+		case "7":
+		case "mox":
+		case "moxie":
+			goal = 7;
+			break;
+		default:
+			return false;
 	}
-	else if((bonus == "hp") || (bonus == "hp regen") || (bonus == "hpregen") || (bonus == "6"))
+	
+	if(contains_text(get_property("_mummeryUses"), goal) || goal < 1 || goal >= 8)
 	{
-		goal = 6;
+		return false;
 	}
-	else if((bonus == "moxie") || (bonus == "7"))
+
+	// CS will use this.
+	if (canChangeFamiliar())
 	{
-		goal = 7;
+		use_familiar(fam);
 	}
 
-	if(contains_text(get_property("_mummeryUses"), goal))
+	cli_execute("mummery " + goal);
+	
+	// CS will use this.
+	if(my_familiar() != last && canChangeFamiliar())
 	{
-		return false;
+		use_familiar(last);
 	}
+	
+	return true;
+}
 
-	if((goal < 1) || (goal >= 8))
+// Will provide the appropriate bonus to an arbitrary familiar.
+boolean mummifyFamiliar(familiar fam)
+{
+	if (!pathHasFamiliar()  || item_amount($item[Mumming Trunk]) == 0 || !auto_is_valid($item[Mumming Trunk]) || !canChangeToFamiliar(fam) || !auto_is_valid(fam))
 	{
 		return false;
 	}
-	use_familiar(fam);
-	string page = visit_url("inv_use.php?pwd=" + my_hash() + "&which=3&whichitem=" + to_int($item[Mumming Trunk]), false);
-	boolean available = false;
 
-	int options = 0;
-	foreach idx in available_choice_options()
-	{
-		if(goal == idx)
-		{
-			available = true;
-		}
-		options += 1;
-	}
-	if(!available)
-	{
-		goal = 8;
-		if(options == 1)
-		{
-			set_property("_mummifyDone", true);
-		}
-	}
-	else if(options == 2)
-	{
-		set_property("_mummifyDone", true);
-	}
-	page = visit_url("choice.php?pwd=&whichchoice=1271&option=" + goal);
+	string targetBonus = "";
 
-	if(my_familiar() != last)
+	switch (my_familiar())
```

Comment:
I'd appreciate feedback on these choices, I realize I'm not even using off-stat buffs at all and there's probably some other good or specific uses I'm not thinking of for these bonuses.